layouts/market.vue

```diff
@@ -0,0 +1,185 @@
+import VSpotMarket from '~/components/partials/spot/market.vue'
+import VDerivativeMarket from '~/components/partials/derivatives/market.vue'
+import VMarketSelection from '~/components/partials/common/markets/markets-selection.vue'
+import VModalMarketBeta from '~/components/partials/modals/market-beta.vue'
+import {
+  ETH_COIN_GECKO_ID,
+  USDT_COIN_GECKO_ID,
+  UST_COIN_GECKO_ID
+} from '~/app/utils/constants'
+import { betaMarketSlugs } from '~/app/data/market'
+import { Modal } from '~/types'
+
+export default Vue.extend({
+  components: {
+    VDerivativeMarket,
+    VMarketSelection,
+    VModalMarketBeta,
+    VSpotMarket
+  },
+
+  props: {
+    overwriteSlug: {
+      type: String,
+      default: undefined
+    }
+  },
+
+  data() {
+    return {
+      status: new Status(StatusType.Loading),
+      isSpotMarket: false,
+      showMarketList: false,
+      interval: 0 as any
+    }
+  },
+
+  computed: {
+    derivativeMarket(): UiDerivativeMarketWithToken | undefined {
+      return this.$accessor.derivatives.market
+    },
+
+    spotMarket(): UiSpotMarketWithToken | undefined {
+      return this.$accessor.spot.market
+    },
+
+    market(): UiSpotMarketWithToken | UiDerivativeMarketWithToken | undefined {
+      const { derivativeMarket, isSpotMarket, spotMarket } = this
+
+      return isSpotMarket ? spotMarket : derivativeMarket
+    },
+
+    slug(): string {
+      const { isSpotMarket, overwriteSlug } = this
+      const { params } = this.$route
+
+      if (overwriteSlug) {
+        return overwriteSlug
+      }
+
+      return isSpotMarket ? params.spot : params.derivative
+    },
+
+    marketIsBeta(): boolean {
+      const { slug } = this
+
+      return betaMarketSlugs.includes(slug)
+    }
+  },
+
+  mounted() {
+    this.isSpotMarket = this.$route.name === 'spot-spot'
+    this.$root.$on('toggle-market-list', this.toggleMarketList)
+
+    Promise.all([this.initMarket(), this.getMarketsQuoteTokenUsdPrices()])
+      .then(() => {
+        this.setMarketQuoteTokenUsdPricesPolling()
+      })
+      .catch(this.$onRejected)
+      .finally(() => {
+        if (this.marketIsBeta) {
+          this.$accessor.modal.openModal(Modal.MarketBeta)
+        }
+
+        this.status.setIdle()
+      })
+
+    Promise.all([
+      this.$accessor.exchange.fetchTradingRewardsCampaign(),
+      this.$accessor.exchange.fetchFeeDiscountAccountInfo()
+    ])
+      .then(() => {})
+      .catch(this.$onRejected)
+  },
+
+  beforeDestroy() {
+    this.$root.$off('toggle-market-list', this.toggleMarketList)
+    this.$accessor.derivatives.reset()
+    this.$accessor.spot.reset()
+    this.$accessor.modal.reset()
+    clearInterval(this.interval)
+  },
+
+  methods: {
+    initMarket(): Promise<void> {
+      const { isSpotMarket, slug } = this
+
+      return isSpotMarket
+        ? this.$accessor.spot.initMarket(slug)
+        : this.$accessor.derivatives.initMarket(slug)
+    },
+
+    getMarketsQuoteTokenUsdPrices(): Promise<void> {
```

Comment:
Why are we doing this?

Reply:
This is to support the total USD volume value for the different trading pairs that we will reflect in the market selection dropdown, for example the `BAYC/WETH perp` pairing.

Reply:
Okay, then we should do it within the `market-selection.vue` component and not in the `layouts/market.vue` component. We should load things within their respective components

Reply:
Good point, will update accordingly in the next PR